DQMOffline/Trigger/plugins/DiJetMonitor.h

```diff
@@ -68,47 +69,70 @@ class DiJetMonitor : public DQMEDAnalyzer
+  void bookME(DQMStore::IBooker &, JetME& me, std::string& histname, std::string& histtitle, int nbinsX, double xmin, double xmax, int nbinsY, double ymin, double ymax);
+  void bookME(DQMStore::IBooker &, JetME& me, std::string& histname, std::string& histtitle, std::vector<double> binningX, std::vector<double> binningY);
+  void setMETitle(JetME& me, std::string titleX, std::string titleY);
   void bookME(DQMStore::IBooker &, MonitorElement* me, std::string& histname, std::string& histtitle, int nbins, double xmin, double xmax);
   void bookME(DQMStore::IBooker &, MonitorElement* me, std::string& histname, std::string& histtitle, int nbinsX, double xmin, double xmax,int nbinsY, double ymin, double ymax );
 
   void analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup) override;
   bool dijet_selection(double eta_1, double phi_1, double eta_2, double phi_2, double pt_1, double pt_2, int &tag_id, int &probe_id);
 
-
 private:
   static MEbinning getHistoPSet    (edm::ParameterSet pset);
   static MEbinning getHistoLSPSet  (edm::ParameterSet pset);
 
   std::string folderName_;
   std::string histoSuffix_;
 
-  edm::EDGetTokenT<reco::PFJetCollection>  dijetSrc_; // test for Jet
-
-  MEbinning           dijetpT_binning;
-
-   DiJetME jetpt1ME_;
-   DiJetME jetpt2ME_;
-   DiJetME jetptAvgaME_;
-   DiJetME jetptAvgbME_;
-   DiJetME jetptTagME_;
-   DiJetME jetptPrbME_;
-   DiJetME jetptAsyME_;
-   DiJetME jetetaPrbME_;
-   DiJetME jetAsyEtaME_;
+  edm::EDGetTokenT<reco::PFMETCollection>       metToken_;
+  edm::EDGetTokenT<reco::PFJetCollection>       pfdijetToken_;// pfjet
+  edm::EDGetTokenT<reco::CaloJetCollection>     calodijetToken_;// calojet
+  edm::EDGetTokenT<reco::GsfElectronCollection> eleToken_;
+  edm::EDGetTokenT<reco::MuonCollection>        muoToken_;
+  edm::EDGetTokenT<reco::PFJetCollection>  dijetSrc_;
+
+  std::vector<double> dijetpT_variable_binning_;
+  MEbinning           dijetpt_binning_;
+  MEbinning           dijetptThr_binning_;
+  MEbinning           ls_binning_;
+
+
+   JetME jetpt1ME_;
+   JetME jetpt2ME_;
+   JetME jetptAvgaME_;
+   JetME jetptAvgaThrME_;
+   JetME jetptAvgbME_;
+   JetME jetptTagME_;
+   JetME jetptPrbME_;
+   JetME jetptAsyME_;
+   JetME jetetaPrbME_;
+   JetME jetetaTagME_;
+   JetME jetphiPrbME_;
+   JetME jetAsyEtaME_;
+   JetME jetEtaPhiME_;
 
   std::unique_ptr<GenericTriggerEventFlag> num_genTriggerEventFlag_;
   std::unique_ptr<GenericTriggerEventFlag> den_genTriggerEventFlag_;
 
+  int nmuons_;
+  double ptcut_;
+  double etacut_;
+  double phicut_;
+
   std::vector<double> v_jetpt;
   std::vector<double> v_jeteta;
   std::vector<double> v_jetphi;
 
+  // Define Phi Bin //
+  double DiJet_MAX_PHI = 3.2;
+  unsigned int DiJet_N_PHI = 64;
+  MEbinning dijet_phi_binning_{
+    DiJet_N_PHI, -DiJet_MAX_PHI, DiJet_MAX_PHI
+  };
   // Define Eta Bin //
   double DiJet_MAX_ETA = 5;
   unsigned int DiJet_N_ETA = 50;
```

Comment:
are you sure you want so many bins ?
wouldn't be better to have a variable binning ?